static/src/js/containers/DownloadHistoryContainer/__tests__/DownloadHistoryContainer.test.js

```diff
@@ -1,89 +1,250 @@
 import {
-  DownloadHistoryContainer,
-  mapDispatchToProps,
-  mapStateToProps
-} from '../DownloadHistoryContainer'
+  render,
+  waitFor,
+  act
+} from '@testing-library/react'
+import { MemoryRouter } from 'react-router-dom'
+import { Provider } from 'react-redux'
+import configureStore from 'redux-mock-store'
+import nock from 'nock'
+
+import { addToast } from '../../../util/addToast'
+import { handleError } from '../../../actions/errors'
+import DownloadHistoryContainer from '../DownloadHistoryContainer'
 import { DownloadHistory } from '../../../components/DownloadHistory/DownloadHistory'
 
-Enzyme.configure({ adapter: new Adapter() })
+jest.mock('../../../util/addToast', () => ({
+  addToast: jest.fn()
+}))
+
+jest.mock('../../../actions/errors', () => ({
+  handleError: jest.fn(() => ({ type: 'HANDLE_ERROR' }))
+}))
+
+let lastPassedOnDeleteRetrieval = null
+
+jest.mock('../../../components/DownloadHistory/DownloadHistory', () => ({
+  DownloadHistory: jest.fn((props) => {
+    lastPassedOnDeleteRetrieval = props.onDeleteRetrieval
+
+    return <div>Download History Component</div>
+  })
+}))
 
-function setup(props) {
-  const enzymeWrapper = shallow(<DownloadHistoryContainer {...props} />)
+const mockStore = configureStore([])
 
-  return {
-    enzymeWrapper,
-    props
-  }
-}
+describe('DownloadHistoryContainer component', () => {
+  let store
 
-describe('mapDispatchToProps', () => {
-  test('onFetchRetrievalHistory calls actions.fetchRetrievalHistory', () => {
-    const dispatch = jest.fn()
-    const spy = jest.spyOn(actions, 'fetchRetrievalHistory')
+  beforeEach(() => {
+    store = mockStore({
+      authToken: 'mock-token',
+      earthdataEnvironment: 'prod'
+    })
 
-    mapDispatchToProps(dispatch).onFetchRetrievalHistory('prod')
+    jest.clearAllMocks()
+    nock.cleanAll()
+    lastPassedOnDeleteRetrieval = null
+  })
 
-    expect(spy).toBeCalledTimes(1)
-    expect(spy).toBeCalledWith('prod')
+  afterEach(() => {
+    nock.cleanAll()
   })
 
-  test('onDeleteRetrieval calls actions.deleteRetrieval', () => {
-    const dispatch = jest.fn()
-    const spy = jest.spyOn(actions, 'deleteRetrieval')
+  describe('when the component renders', () => {
+    test('renders the DownloadHistory component with the correct props', async () => {
+      const historyData = [{
+        id: '8069076',
+        jsondata: {},
+        created_at: '2019-08-25T11:58:14.390Z',
+        collections: [{}]
+      }]
+      nock(/.*/)
+        .get(/retrievals/)
+        .reply(200, historyData)
 
-    mapDispatchToProps(dispatch).onDeleteRetrieval('retrievalId')
+      render(
+        <Provider store={store}>
+          <MemoryRouter>
+            <DownloadHistoryContainer />
+          </MemoryRouter>
+        </Provider>
+      )
 
-    expect(spy).toBeCalledTimes(1)
-    expect(spy).toBeCalledWith('retrievalId')
+      await waitFor(() => {
+        expect(DownloadHistory).toHaveBeenLastCalledWith(
```

Comment:
Everywhere you test a mock you have to test how many times it has been called (`toHaveBeenCalledTimes()`) along with it being called with the correct parameters. That tells you how many times the component has been rendered, which can be an indicator of problems